aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,8 +3429,30 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if (driver == "gelss" && input.device() != at::kCPU) {
+    if (input.numel() == 0 || input.size(-2) == 0 || input.size(-1) == 0) {
+        auto output_shape = input.sizes().vec();
+        output_shape.back() = other.size(-1);
```

Comment:
This is not correct. But even more, `solution` has already been allocated, you just need to zero it out (for the case where it may not be empty)